File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOWatchDialSortParamModel.swift
//
//  IDOWatchDialSortParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/18.
//

import Foundation

@objcMembers
public class IDOWatchDialSortParamModel: NSObject, IDOBaseModel {
    /// Number of items in the watch dial sort list
    public var sortItemNumb: Int
    /// Array of watch dial sort items type, sort_number, and name
    public var pSortItem: [IDOWatchDialSortItem]

    enum CodingKeys: String, CodingKey {
        case sortItemNumb = "sort_item_numb"
        case pSortItem = "p_sort_item"
    }

    public init(sortItemNumb: Int, pSortItem: [IDOWatchDialSortItem]) {
        self.sortItemNumb = sortItemNumb
        self.pSortItem = pSortItem
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

// MARK: - IDOWatchDialSortItem
@objcMembers
public class IDOWatchDialSortItem: NSObject, Codable {
    /// Watch dial type 1: Normal Dial, 2: Wallpaper Dial, 3: Cloud Dial
    public var type: Int
    /// Serial number, starting from 0, not exceeding the total number of supported watch dials
    public var sortNumber: Int
    /// Watch dial ID, maximum 29 bytes
    public var name: String

    enum CodingKeys: String, CodingKey {
        case type
        case sortNumber = "sort_number"
        case name
    }

    public init(type: Int, sortNumber: Int, name: String) {
        self.type = type
        self.sortNumber = sortNumber
        self.name = name
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOWatchDialSortParamModel.kt
//
//  IDOWatchDialSortParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import java.io.Serializable
import com.google.gson.annotations.SerializedName

///

open class IDOWatchDialSortParamModel(sortItemNumb: Int, pSortItem: List<IDOWatchDialSortItem>) :
    IDOBaseModel {

    /// Number of items in the watch dial sort list
    @SerializedName("sort_item_numb")
    var sortItemNumb: Int = sortItemNumb

    /// Array of watch dial sort items type, sort_number, and name
    @SerializedName("p_sort_item")
    var pSortItem: List<IDOWatchDialSortItem> = pSortItem

    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}

// MARK: - IDOWatchDialSortItem
open class IDOWatchDialSortItem(type: Int, sortNumber: Int, name: String) : Serializable {
    /// Watch dial type 1: Normal Dial, 2: Wallpaper Dial, 3: Cloud Dial
    @SerializedName("type")
    var type: Int = type

    /// Serial number, starting from 0, not exceeding the total number of supported watch dials
    @SerializedName("sort_number")
    var sortNumber: Int = sortNumber

    /// Watch dial ID, maximum 29 bytes
    @SerializedName("name")
    var name: String = name
}